internal/monitors/processlist/processlist_windows.go

```diff
@@ -0,0 +1,172 @@
+	return ps, err
+}
+
+// getUsername - retrieves a username from an open process handle it is set as a package variable so we can mock it durring testing
+var getUsername = func(id uint32) (username string, err error) {
+	// open the process handle and collect any information that requires it
+	var h windows.Handle
+	defer windows.CloseHandle(h)
+	if h, err = windows.OpenProcess(processQueryLimitedInformation, false, id); err != nil {
+		err = fmt.Errorf("unable to open process handle. %v", err)
+		return username, err
+	}
+
+	// the windows api docs suggest that windows.TOKEN_READ is a super set of windows.TOKEN_QUERY,
+	// but in practice windows.TOKEN_READ seems to be less permissive for the admin user
+	var token windows.Token
+	defer token.Close()
+	err = windows.OpenProcessToken(h, windows.TOKEN_QUERY, &token)
+	if err != nil {
+		err = fmt.Errorf("unable to retrieve process token. %v", err)
+		return username, err
+	}
+
+	// extract the user from the process token
+	user, err := token.GetTokenUser()
+	if err != nil {
+		err = fmt.Errorf("unable to get token user. %v", err)
+		return username, err
+	}
+
+	// extract the username and domain from the user
+	userid, domain, _, err := user.User.Sid.LookupAccount("")
+	if err != nil {
+		err = fmt.Errorf("unable to look up user account from Sid. %v", err)
+	}
+	username = fmt.Sprintf("%s\\%s", domain, userid)
+
+	return username, err
+}
+
+// ProcessList takes a snapshot of running processes and returns a byte buffer
+func ProcessList() (*bytes.Buffer, error) {
+	processes := &bytes.Buffer{}
+	processes.WriteString("{")
+	defer processes.WriteString("}") // always close the associative array
+
+	// Get all processes
+	ps, err := getAllProcesses()
+	if err != nil {
+		logger.Debugf("no processes returned %v", err)
+		return processes, err
+	}
+
+	// Get cpu percentages for all processes
+	cpuPercentages, err := getCPUPercentages()
+	if err != nil {
+		logger.Debugf("no per process cpu percentages returned %v", err)
+	}
+
+	// index position to stop appending commas to the list of processes
+	stop := len(ps) - 1
+
+	// iterate over each process and build an entry for the process list
+	for index, p := range ps {
+		username, err := getUsername(p.ProcessID)
+		if err != nil {
+			logger.Debugf("unable to collect use name for process %v %v", p, err)
+		}
+
+		// CPU Times
+		var cpuPercent float64
+		totalTime := float64(p.UserModeTime+p.KernelModeTime) / 10000000 // 100 ns units to seconds
+		if percent, inMap := cpuPercentages[p.ProcessID]; inMap {
+			cpuPercent = float64(percent)
+		}
+
+		// Memory Percent
+		var memPercent float64
+		if systemMemory, err := mem.VirtualMemory(); err == nil {
+			memPercent = 100 * float64(p.WorkingSetSize) / float64(systemMemory.Total)
+		} else {
+			logger.Debugf("unable to collect system memory total", err)
```

Comment:
no `%s` in the format string to stick the err in.  Also doesn't this warrant an error message?